### PR TITLE
[Improve] remove delete user function and add transfer resource of user function

### LINKED ISSUE #2712: [Bug] job cannot be displayed
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-streampark/issues?q=is%3Aissue+label%3A%22bug%22) and found no similar issues.

### Java Version

1.8

### Scala Version

2.12.x

### StreamPark Version

2.0.0

### Flink Version

1.14.4

### deploy mode

None

### What happened

Background: Someone resigned from the company and the administrator deleted the streampark user.
Impact: Tasks created by deleted users cannot be displayed in the application management interface.

### Error Exception

```log
nothing
```

### Screenshots

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!(您是否要贡献这个PR?)

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by VampireAchao**
## What changes were proposed in this pull request

Issue Number: close #2712 

## Brief change log

1. Requirement:

- The administrator cancels the user deletion function and replaces it with disabling users.

image

2. Detailed logic:

- The administrator cancels the user deletion function.
- When the user account is disabled, it cannot own resources (application and project ownership)
- Add resource transfer function to transfer disabled user resources to new users

8f04aebad699a2e235467432a89f976

## Verifying this change

Change the user's status when the user has resources.

This change is already covered by existing tests, such as *UserServiceTest*.

This change added tests and can be verified as follows:

1. testLockUser in UserServiceTest
- This unit test covers different scenarios for locking and unlocking a user when they have no resources, as well as locking a user when they have resources and checking the different return results.
3. testTransferResource in UserServiceTest
- This unit test involves transferring a user's resources to another person.

## Does this pull request potentially affect one of the following parts
 - Dependencies (does it add or upgrade a dependency): (yes / no)
 no

### CODE REVIEW

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/ApplicationService.java [hunk 1]
```diff
@@ -120,4 +122,6 @@ List<Application> getByTeamIdAndExecutionModes(
   List<String> historyUploadJars();
 
   String k8sStartLog(Long id, Integer offset, Integer limit) throws Exception;
+
+  void changeUser(Long userId, Long targetUserId);
```
Comment (wolfboys) [inline streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/ApplicationService.java]: It is recommended to change the method name to `changeOwnership`

File: streampark-console/streampark-console-webapp/src/views/system/user/components/UserDrawer.vue [hunk 2]
```diff
@@ -28,24 +28,45 @@
...
+  <Modal
+    :visible="transferModalVisible"
+    :confirm-loading="transferModalLoading"
+    :ok-text="t('common.okText')"
+    centered
+    @ok="handleTransfer"
+    @cancel="transferModalVisible=false"
+  >
+    <template #title>
+      <SvgIcon name="alarm" size="25"/>
```
Comment (wolfboys) [inline streampark-console/streampark-console-webapp/src/views/system/user/components/UserDrawer.vue]: It is recommended change icon to  `<Icon icon="ant-design:swap-outlined" />`

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/ApplicationService.java (hunk 1 above)
Comment (VampireAchao) [reply streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/ApplicationService.java]: okay

**Comment by VampireAchao** [comment]
I will modify the modal

**Review by wolfboys — APPROVED** [review]
LGTM

**Pull request merged**